src/transformers/models/clip/modeling_tf_clip.py

```diff
@@ -0,0 +1,1528 @@
+            input_ids=input_ids,
+            pixel_values=pixel_values,
+            attention_mask=attention_mask,
+            position_ids=position_ids,
+            return_loss=return_loss,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            training=training,
+            kwargs_call=kwargs,
+        )
+
+        if inputs["input_ids"] is None:
+            raise ValueError("You have to specify either input_ids")
+        if inputs["pixel_values"] is None:
+            raise ValueError("You have to specify pixel_values")
+
+        input_shape = shape_list(inputs["input_ids"])
+
+        if inputs["attention_mask"] is None:
+            inputs["attention_mask"] = tf.fill(dims=input_shape, value=1)
+
+        vision_outputs = self.vision_model(
+            pixel_values=inputs["pixel_values"],
+            output_attentions=inputs["output_attentions"],
+            output_hidden_states=inputs["output_hidden_states"],
+            return_dict=inputs["return_dict"],
+            training=inputs["training"],
+        )
+
+        text_outputs = self.text_model(
+            input_ids=inputs["input_ids"],
+            attention_mask=inputs["attention_mask"],
+            position_ids=inputs["position_ids"],
+            output_attentions=inputs["output_attentions"],
+            output_hidden_states=inputs["output_hidden_states"],
+            return_dict=inputs["return_dict"],
+            training=inputs["training"],
+        )
+
+        image_embeds = vision_outputs[1]
+        image_embeds = self.visual_projection(inputs=image_embeds)
+
+        text_embeds = text_outputs[1]
+        text_embeds = self.text_projection(inputs=text_embeds)
+
+        # normalized features
+        image_embeds = image_embeds / tf.norm(tensor=image_embeds, ord="euclidean", axis=-1, keepdims=True)
+        text_embeds = text_embeds / tf.norm(tensor=text_embeds, ord="euclidean", axis=-1, keepdims=True)
+
+        # cosine similarity as logits
+        logit_scale = tf.math.exp(self.logit_scale)
+        logits_per_text = tf.matmul(text_embeds, image_embeds, transpose_b=True) * logit_scale
+        logits_per_image = tf.transpose(logits_per_text)
+
+        loss = None
+        if inputs["return_loss"]:
+            loss = clip_loss(logits_per_text)
+
+        if not inputs["return_dict"]:
+            output = (logits_per_image, logits_per_text, text_embeds, image_embeds, text_outputs, vision_outputs)
+            return (loss,) + output if loss is not None else output
+
+        return TFCLIPOutput(
+            loss=loss,
+            logits_per_image=logits_per_image,
+            logits_per_text=logits_per_text,
+            text_embeds=text_embeds,
+            image_embeds=image_embeds,
+            text_model_output=text_outputs,
+            vision_model_output=vision_outputs,
+        )
+
+
+class TFCLIPPreTrainedModel(TFPreTrainedModel):
```

Comment:
The PyTorch version of this class also included [the weight initializations for all of the layers in the model](https://github.com/huggingface/transformers/blob/master/src/transformers/models/clip/modeling_clip.py#L345-L386) - I agree that this probably isn't the right place to put them in TF, but we should make sure that we use the same initialization values.